### PR TITLE
refactor: tests

### PR DESCRIPTION
**Pull request by Daanvdplas**

### CODE REVIEW

File: pallets/nfts/src/tests.rs [hunk 1]
```diff
@@ -2604,136 +2606,103 @@ fn approve_collection_transfer_works() {
...
-				delegate: delegate.clone(),
-				deadline: None,
-			}));
-			assert_eq!(Balances::reserved_balance(&item_owner), 1);
 
-			// Approve same delegate again not updating the total reserved funds.
-			assert_ok!(approve_collection_transfer(
-				origin.clone(),
-				maybe_item_owner.clone(),
+			for deadline in [None, None, Some(deadline), Some(deadline * 2), Some(deadline), None] {
```
Comment (chungquantin) [inline pallets/nfts/src/tests.rs:2654]: Sorry but I don't understand what this means.

File: pallets/nfts/src/tests.rs [hunk 2]
```diff
@@ -2415,55 +2424,47 @@ fn cancel_collection_approval_works() {
...
-					account(4)
-				),
-				Error::<Test>::NoPermission
+			assert!(!CollectionApprovals::contains_key((collection_id, &item_owner, &delegate)));
+			System::assert_last_event(
+				Event::<Test>::ApprovalCancelled {
+					collection: collection_id,
+					item: None,
+					owner: item_owner.clone(),
+					delegate: delegate.clone(),
```
Comment (chungquantin) [inline pallets/nfts/src/tests.rs:2465]:
```suggestion
					delegate,
```

File: pallets/nfts/src/tests.rs [hunk 3]
```diff
@@ -2415,55 +2424,47 @@ fn cancel_collection_approval_works() {
...
-					item_id,
-					account(4)
-				),
-				Error::<Test>::NoPermission
+			assert!(!CollectionApprovals::contains_key((collection_id, &item_owner, &delegate)));
+			System::assert_last_event(
+				Event::<Test>::ApprovalCancelled {
+					collection: collection_id,
+					item: None,
+					owner: item_owner.clone(),
```
[the remark this answers is not in the document]
Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs:2464]: It is in a for loop

File: pallets/nfts/src/tests.rs (hunk 2 above)
Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs:2465]: for loop

File: pallets/nfts/src/tests.rs (hunk 1 above)
Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs:2657]: see latest commit

File: pallets/nfts/src/tests.rs [hunk 4]
```diff
@@ -2604,136 +2606,106 @@ fn approve_collection_transfer_works() {
...
 
-			// Approve same delegate again not updating the total reserved funds.
-			assert_ok!(approve_collection_transfer(
-				origin.clone(),
-				maybe_item_owner.clone(),
+			// Approving a collection with no deadline (`None`) and a deadline. The repetition is to
+			// demonstrate all cases possible (e.g. approving a delegate with no deadline to with a
+			// deadline).
+			for deadline in [None, None, Some(deadline), Some(deadline * 2), Some(deadline), None] {
+				// Approve same delegate with no deadline.
```
Comment (chungquantin) [inline pallets/nfts/src/tests.rs:2658]:
```suggestion
```
With the above changes, this comment is deprecated.

File: pallets/nfts/src/tests.rs (hunk 1 above)
Comment (chungquantin) [reply pallets/nfts/src/tests.rs:2657]: This is sadly still not easy to understand.  Can we use `System::set_block_number(0)` to reset the time and update the list of deadlines to fit with that?

File: pallets/nfts/src/tests.rs [hunk 5]
```diff
@@ -3095,131 +3067,108 @@ fn clear_collection_approvals_works() {
...
+				.is_zero());
+			assert!(events().contains(&Event::<Test>::ApprovalsCancelled {
 				collection: collection_id,
 				item: None,
-				owner: owner.clone(),
+				owner: item_owner.clone(),
 			}));
-
-			assert_noop!(
-				Nfts::transfer(
```
Comment (chungquantin) [inline pallets/nfts/src/tests.rs:3206]: just let you know that this follows the behaviour of the `clear_all_transfer_approvals` test.

**Review by chungquantin — CHANGES_REQUESTED** [review]
Only the deadline for loop is still confusing to me even with a description. Maybe it could be better to explain each case, I guess.

File: pallets/nfts/src/tests.rs (hunk 5 above)
Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs:3206]: Not necessary as the `CollectionApprovals` storage item is 0 and tested.

Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs:3206]: Added it anyway :)

File: pallets/nfts/src/tests.rs (hunk 1 above)
Comment (Daanvdplas) [reply pallets/nfts/src/tests.rs]: See latest commit

**Review by chungquantin — APPROVED** [review]

**Pull request merged**